### ISSUE TITLE
Incompatible with REST API for signal (needed for compliance with WCAG & Americans with Disabilities Act)

### ISSUE THREAD

**Issue by jaanli** [mautrix/signal#521]
Hi! I need to use screen reading for my disability (see the ADA for more details or WCAG), and for translation.

For this I use this Signal REST API for most content: 

https://github.com/bbernhard/signal-cli-rest-api

Can you please help me understand what pull request would be needed in order to fix the issue and enable local LLMs to be used for screen reading and summarization to bring this into compliance? 

Thank you!
Jaan

P.S. We have some resources @onefact for supporting this type of work as well, please let me know if that may be of interest - that is because @signalapp is one of the few HIPAA-compliant consumer oriented solutions we use for the hospitals and clinics we work with to share protected health information in compliance with federal law.

**Comment by tulir** [comment]
You forgot to specify what the issue is

**Comment by tulir** [comment]
I'm not sure how that's related to the bridge, but in general all Beeper issues should be reported via the in-app "Report a problem" button rather than github

**Comment by jaanli** [comment]
Ah sorry for being unclear @tulir , the bridge issue is the REST API - we cannot integrate it with https://github.com/bbernhard/signal-cli-rest-api in order to send and receive messages from large language models running on AWS. 

Does that make more sense? I'm happy to send over a UML diagram or other format (Figma?) if it is easier to see our architecture.

**Comment by tulir** [comment]
The bridge works just fine with signal-cli as the primary device. When logging in via commands, the bot will send the raw linking URI (`sgnl://...`) in addition to the QR image, which can be passed to signal-cli's `addDevice` command. `POST /v1/devices/{number}` with `uri` in the body appears to be the equivalent method in that API.

If you meant something else than linking, it's probably out of scope for the bridge. The bridge is only for bridging messages between Matrix and Signal. Sending and receiving messages has to be done on Matrix or Signal via their respective APIs, not in the bridge.

**Comment by jaanli** [comment]
Ah got it, sorry I thought the bridge was for bridging messages between Matrix, Signal, and large language models. 

I think I was mistaken so I will close the issue shortly - my bad!

Would a new bridge be needed to bridge messages between Matrix and large language model servers? 

(Whether language models are used locally, in WebAssembly e.g. @WasmEdge - https://github.com/LlamaEdge/LlamaEdge - or in the cloud, like on @aws Lightsail)

**Comment by tulir** [comment]
LLM integrations would probably either be bots (like https://github.com/matrixgpt/matrix-chatgpt-bot) or something integrated into clients (like https://github.com/beeper/widget-summarizer)

**Comment by jaanli** [comment]
> The bridge works just fine with signal-cli as the primary device. When logging in via commands, the bot will send the raw linking URI (`sgnl://...`) in addition to the QR image, which can be passed to signal-cli's `addDevice` command. `POST /v1/devices/{number}` with `uri` in the body appears to be the equivalent method in that API.
> 
> If you meant something else than linking, it's probably out of scope for the bridge. The bridge is only for bridging messages between Matrix and Signal. Sending and receiving messages has to be done on Matrix or Signal via their respective APIs, not in the bridge.

Thank you so much @tulir ! That is exactly what we need - we are building physical devices to help people create bridges via the API & bridge system you described.

We will keep you posted if we run into any issues.
